src/renderers/webgl/node.ts

```diff
@@ -1,23 +1,25 @@
-import { DEFAULT_LABEL_STYLE, MIN_LABEL_ZOOM, MIN_INTERACTION_ZOOM, MIN_NODE_STROKE_ZOOM, MIN_NODE_ICON_ZOOM } from '../../utils/constants'
+import { DEFAULT_LABEL_STYLE, MIN_LABEL_ZOOM, MIN_INTERACTION_ZOOM, MIN_STROKE_ZOOM, MIN_NODE_ICON_ZOOM } from '../../utils/constants'
 import { FederatedPointerEvent } from 'pixi.js'
-import { NodeStrokes } from './objects/nodeStrokes'
 import { NodeHitArea } from './interaction/nodeHitArea'
 import { interpolate } from '../../utils/helpers'
-import { NodeFill } from './objects/nodeFill'
 import { type Renderer } from '.'
 import type { Node } from '../../types'
+import Circle from './objects/circle/Circle'
+import CircleStrokes from './objects/circle/CircleStrokes'
 import Text from './objects/text/Text'
 import Icon from './objects/Icon'
 
 export class NodeRenderer {
-  node!: Node
-  x!: number
-  y!: number
-  fill: NodeFill
+  x = 0
+  y = 0
+
+  node: Node
+  fill: Circle
+  strokes?: CircleStrokes
```

Comment:
conditionally rendering strokes until an object is visible is a perf boost. should consider doing the same for fills